File: top-interview-questions-medium/Backtracking/subsets/78.py
class Solution:
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        if not len(nums): return []
        if len(nums) == 1: return [[], [nums[0]]]
        last = self.subsets(nums[:-1])
        return last + list(map(lambda v: v + nums[-1:], last))

File: top-interview-questions-medium/Backtracking/subsets/78.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
  if (!nums.length) return []
  if (nums.length === 1) return [[], [nums[0]]]
  const last = subsets(nums.slice(0, -1))
  return last.concat(last.map(v => v.concat(nums.slice(-1))))
}
